### ISSUE TITLE
Job submitter bottleneck

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#8129]
Let me put this in a GH issue as I think this is going (was) to be impacting protection.
With large number of low priority (63k prio) pending on all agents, higher priority work WQE does get acquired, the jobs are created (regardless of thresholds according to @amaltaro ) but then jobs do not get submitted over threshold.
Example an agent with 40k job pending threshold could be holding 35k low priority job and de-facto is reduced to a 5k effective threshold for all the rest. That is, the low priority pending have no chance at running (because of higher priority work being ran otherwise from other agents) and will not free their pending slot, leaving very little room for other work that would deserve to get submitted and ran.

we should find a way to solve this

**Comment by vlimant** [comment]
example. Once I force-completed https://cmsweb.cern.ch/reqmgr2/fetch?rid=mcremone_task_EXO-RunIISummer15wmLHEGS-04804__v1_T_170809_021622_8251

![mcremone_task_exo-runiisummer15wmlhegs-04804__v1_t_170809_021622_8251](https://user-images.githubusercontent.com/4989875/30112672-6d49f240-9312-11e7-9490-e1807f61b9d8.png)

many other wokflows got plenty of (already created jobs) jobs submitted

![fabozzi_run2016f-v1-charmonium-07aug17_8029_170831_193216_1575](https://user-images.githubusercontent.com/4989875/30112701-824a3a4c-9312-11e7-80ad-93cfc416f5d2.png)

this makes priority inversion by starving from higher priority work IMO

**Comment by ticoann** [comment]
I think we discussed before. 
Current system works like following
1, HIgh priority work will still get to Agent even if there are more jobs then the threshold (threshold proportional to pending threshold). 
2. However it won't get submitted already submitted jobs are more than pending threshold.

We could submit more jobs even if pending threshold is met. But that could cause other problems (i.e. pending jobs could be indefinitely grow)

I think the problem occurs we remove a bunch of jobs from condor in short time period, then before new jobs gets to be submitted, many jobs (lower priority) gets to be submitted.

Maybe in this particular case, we can slow down the removal of jobs by diving small numbers periodically, then there will be more (high priority) jobs submitted and minimize the problem.

**Comment by vlimant** [comment]
Indeed the number of job submitted could grow "indefinitely", if higher and higher priority work gets in the system, but that's what we need, submit the higher priority work.
There might be the need for removing jobs sooner, with a mechanism where the agent keep the amount of pending work constant, by removing N from the bottom, when submitting N from the top.
Other possibility would be that the agent is a pass-through (all acquired gets submitted right away) and the thresholds are lowered enough so that an agent never has too deep of a queue.
accounting in number of CPU instead of jobs would greately help in thinking of a more sustainable solution.

**Comment by vlimant** [comment]
see these guys https://dmytro.web.cern.ch/dmytro/cmsprodmon/requests.php?in_production=1&campaign=Commissioning2017 being held because of this at the moment.
can try to figure out removing some low priority jobs from the agent.

**Comment by vlimant** [comment]
I restarted components in 253 with higher thresholds to let go of jobs that spend the week-end in "created". That did the trick

**Comment by vlimant** [comment]
@amaltaro @ticoann this is getting critical. Is there any way we can mitigate this ?

**Comment by ticoann** [comment]
I am not still sure it is good way to solve this with putting higher priority in Glidein regardless with pending threshold. Didn't we also have problem with too many pending jobs in the system. #8224 
I will discuss with Alan and try to find some solution.
